edgefirewall.yaml

```diff
@@ -518,12 +549,66 @@ components:
           type: boolean
 
     Behaviors:
+      type: object
       properties:
         name:
           type: string
           enum: [deny, drop, set_rate_limit, set_waf_ruleset, run_function, tag_event]
-        target:
+        argument:
+          oneOf:
+            - $ref: '#/components/schemas/SetRateLimitDetails'
+            - $ref: '#/components/schemas/SetWAFRuleSetDetails'
```

Comment:
**suggestion:** we may need to add a generic Behavior as we not defined the specific schema to all other possible behaviors

```suggestion
            - $ref: '#/components/schemas/SetWAFRuleSetDetails'
            - $ref: '#/components/schemas/GenericBehaviorDetails'  # TBD
```

Reply:
We define both complex details(rate_limit and waf).
Why do we need to provide the simple one?
It just a behavior with the name property.

Reply:
We didn't block any behavior name, when we use set_rate_limit or set_waf_ruleset we have a specific detail, that's all.

Reply:
I think a got your point... the key word is "oneOf" - you are right!

Reply:
Sorry, I arrived here too late. But you got the point.